### DISCUSSION TITLE
Can't start service without sudo

### DISCUSSION THREAD

**Discussion by omnigenous** [davmac314/dinit#265] [Q&A] [answered]
I need to run `syncthing` service as non-root because it's giving me warning:

"Syncthing should not run as a privileged or system user. Please consider using a normal user account."

And I can't access folder with all sync files with root.

I tried disabling service with root:
`sudo dinitctl disable syncthing`

Then run:
`dinitctl enable syncthing`
And I get this error:
"dinitctl: connecting to socket /home/USER/.dinitctl: No such file or directory"

Then created empty file:
`touch ~/.dinitctl`
Moved syncthing service:
`mkdir ~/.config/dinit.d`
`sudo mv etc/dinit.d/syncthing ~/.config/dinit.d`
`dinitctl start syncthing`
"dinitctl: connecting to socket /home/USER/.dinitctl: Connection refused"

So I didn't know what to put into that file and I looked at man pages saw only one mention of `~/.dinitctl` and root equivalent would be `/etc/dinitctl` which doesn't exist on my system.

I saw #183 and #189 but it seems I got different issue and it could have been something I messed up while setting up dinit during install.
Is there step-by-step guide how to get dinitctl working without root?

**Comment by mobin-2008** [comment, marked as answer]
Hi.

I think you misunderstood about using dinit as service manager.
For running a service as any user than root you have these options:

1. Running your service in system instance of dinit. You should create a service for system instance of dinit in `/etc/dinit.d/` or `/lib/dinit.d/` or `/usr/local/lib/dinit.d` directory and set `run-as` option as your user:
    ```bash
    command = /usr/bin/my_daemon
    run-as = mobin
    ...
    ```

2. Running a user instance of dinit and put your service on it. You should create `~/.config/dinit.d` and create `boot` service at threre and run dinit as your user by `dinit &` command. then put your services on `~/.config/dinit.d` directory and start them with dinitctl without `sudo` and as your user: `dinitctl start my_service`. Full guide at https://github.com/davmac314/dinit/blob/master/doc/getting_started.md

Also dinit daemon for your user will be not ran by default and you need to run it manually. Take a look at #183

Regards

**Reply by omnigenous** [reply]
Thank you so much for detailed comment! I went with option 1 and edited `/etc/dinit.d/syncthing` file `run-as` line. I will read docs you linked to get better understanding once I get some sleep :D

**Reply by mobin-2008** [reply]
Your welcome 😁
